src/pykeen/checkpoints/__init__.py

```diff
@@ -0,0 +1,116 @@
+"""
+This module contains methods for deciding when to write and clear checkpoints.
+
+.. warning ::
+    While this module provides a flexible and modular way to describe a desired checkpoint behavior, it currently only
+    stores the model's weights (more precisely, its :meth:`torch.nn.Module.state_dict`).
+    Thus, it does not yet replace the full training loop checkpointing mechanism described in
+    :ref:`regular_checkpoints_how_to`.
+
+It consists of two main components: checkpoint *schedules* decide whether to write a checkpoint at a given epoch.
+If we have multiple checkpoints, we can use multiple *keep strategies* to decide which checkpoints to keep and
+which to discard. For both, we provide a set of basic rules, as well as a way to combine them via union.
+Those should be sufficient to easily model most of the desired checkpointing behaviours.
+
+Examples
+========
+
+Below you can find a few examples of how to use them inside the training pipeline.
+If you want to check before an actual training how (static) checkpoint schedules behave,
+you can take a look at :meth:`pykeen.checkpoints.final_checkpoints`
+and :meth:`pykeen.checkpoints.inspect_schedule`.
+
+Example 1
+~~~~~~~~~
+Write a checkpoint every 10 steps and keep them all.
+
+.. code-block::
+
+    from pykeen.pipeline import
+
+    result = pipeline(
+        dataset="nations",
+        model="mure",
+        training_kwargs=dict(
+            num_epochs=100,
+            callbacks="checkpoint",
+            # create one checkpoint every 10 epochs
+            callbacks_kwargs=dict(
+                schedule="every",
+                schedule_kwargs=dict(
+                    frequency=10,
+                ),
+            )
+        ),
+    )
+
+Example 2
+~~~~~~~~~
+Write a checkpoint every 10 steps, but keep only the last one and one every 50 steps.
+
+.. code-block::
+
+    from pykeen.pipeline import
+
+    result = pipeline(
+        dataset="nations",
+        model="mure",
+        training_kwargs=dict(
+            num_epochs=100,
+            callbacks="checkpoint",
+            # create one checkpoint every 10 epochs
+            callbacks_kwargs=dict(
+                schedule="every",
+                schedule_kwargs=dict(
+                    frequency=10,
+                ),
+                keeper="union",
+                keeper_kwargs=dict(
+                    bases=["modulo", "last"],
+                    bases_kwargs=[dict(divisor=50), None],
+                )
+            )
+        ),
+    )
```

Comment:
can you give an example using the more complicated one with the best checker? most likely since someone will read the high level docs but not necessarily go into the specific ones

Reply:
https://github.com/pykeen/pykeen/pull/1430/commits/2dc92993d696c754d49b337ef29028c858299ec8